include/grpcpp/impl/codegen/client_callback.h

```diff
@@ -157,28 +157,95 @@ class ClientCallbackWriter {
   }
 };
 
-// The user must implement this reactor interface with reactions to each event
-// type that gets called by the library. An empty reaction is provided by
-// default
+// The following classes are the reactor interfaces that are to be implemented
+// by the user. They are passed in to the library as an argument to a call on a
+// stub (either a codegen-ed call or a generic call). The streaming RPC is
+// activated by calling StartCall, possibly after initiating StartRead,
+// StartWrite, or AddHold operations on the streaming object. Note that none of
+// the classes are pure; all reactions have a default empty reaction so that the
+// user class only needs to override those classes that it cares about.
+
+/// \a ClientBidiReactor is the interface for a bidirectional streaming RPC.
 template <class Request, class Response>
 class ClientBidiReactor {
  public:
   virtual ~ClientBidiReactor() {}
+
+  /// Notifies the application that all operations associated with this RPC
+  /// have completed and provides the RPC status outcome.
+  ///
+  /// \param[in] s The status outcome of this RPC
   virtual void OnDone(const Status& s) {}
+
+  /// Notifies the application that a read of initial metadata from the
+  /// server is done. If the application chooses not to implement this method,
+  /// it can assume that the initial metadata has been read before the first
+  /// call of OnReadDone or OnDone.
+  ///
+  /// \param[in] ok Was the initial metadata read successfully? If false, no
+  ///               further read-side operation will succeed.
   virtual void OnReadInitialMetadataDone(bool ok) {}
+
+  /// Notifies the application that a Read operation completed.
+  ///
+  /// \param[in] ok Was it successful? If false, no further read-side operation
+  ///               will succeed
   virtual void OnReadDone(bool ok) {}
+
+  /// Notifies the application that a Write operation completed.
+  ///
+  /// \param[in] ok Was it successful? If false, no further write-side operation
+  ///               will succeed
   virtual void OnWriteDone(bool ok) {}
+
+  /// Notifies the application that a WritesDone operation completed. Note that
+  /// this is only used on explicit WritesDone operations and not for those that
+  /// are implicitly invoked as part of a StartWriteLast.
+  ///
+  /// \param[in] ok Was it successful?
```

Comment:
There's nothing that you can do or derive at this point since there aren't any valid write-side operations that you can execute after the WritesDone in any case. I guess I should comment that WritesDone (or equivalents) can be initiated at most once for an RPC.

Reply:
Added some info and restrictions.